### PR TITLE
162262078 transport operator ytj merge section

### PR DESCRIPTION
**Pull request by philosotractor**
# Added
* some themed colours
* re-usable components for displaying a themed icon-lable, e.g. for conveying a result next to an UI control
* (re-usable) styling for several UI components
* transport-operator (ytj) operator name merging wizard section
* localisation strings with initial values
   
# Changed

# fixed
* test: fix admin_test to evaluate last auditlog event, not last
* ui: select-field support for dynamically changing options
* controller: transport-operator remove debug logs

### CODE REVIEW

File: ote/resources/public/language/sv.edn [hunk 1]
```diff
@@ -338,6 +338,7 @@
  :buttons
  {:save "Spara"
   :cancel "Ångra"
+  :continue "Fortsätta"
```
Comment (rckopteri) [inline ote/resources/public/language/sv.edn]: Could you add these text to this ticket: https://www.pivotaltracker.com/story/show/163085064

File: ote/src/clj/ote/services/operators.sql [hunk 2]
```diff
@@ -38,4 +38,10 @@ SELECT u.id as "user-id"
...
    AND u.id != :user-id
- GROUP BY "user-id";
\ No newline at end of file
+ GROUP BY "user-id";
+ 
+-- name: group-id-for-op
+SELECT "ckan-group-id" FROM "transport-operator" op
+ WHERE op."id" = :id
+ LIMIT 1;
+>>>>>>> services: resolve group id when updating transport-operator
```
Comment (rckopteri) [inline ote/src/clj/ote/services/operators.sql]: conflict message

File: ote/src/clj/ote/services/transport.clj [hunk 3]
```diff
@@ -279,6 +279,14 @@
                 ::t-operator/email
                 ::t-operator/homepage]))
 
+;; Takes db and transport-operator. Resolves optional data for updating operator and returns a transport-operator map.
+(defn resolve-update-operator-data [db op]
+  (let [ckan-id (when (nil? (::t-operator/ckan-group-id op))
+                  (group-id-for-op db {:id (::t-operator/id op)}))]
+    ;; cond used just in case there are mode fields to handle in the future
+    (cond-> op
+            (some? ckan-id) (assoc ::t-operator/ckan-group-id (:ckan-group-id (first ckan-id))))))
```
Comment (rckopteri) [inline ote/src/clj/ote/services/transport.clj]: Could you move this assoc to the when clause and return op.

File: ote/resources/public/language/sv.edn (hunk 1 above)
Comment (philosotractor) [reply ote/resources/public/language/sv.edn]: Sure, I'm about to create a new ticket about all localisations :)

File: ote/src/clj/ote/services/operators.sql (hunk 2 above)
Comment (philosotractor) [reply ote/src/clj/ote/services/operators.sql]: Ouch!

File: ote/src/cljs/ote/views/transport_operator_ytj.cljs [hunk 4]
```diff
@@ -261,7 +266,50 @@
...
+
+(defn- operator-merge-section [e! {nap-orphans :ytj-orphan-nap-operators ops-to-save :transport-operators-to-save :as operator} ytj-company-names app]
+  [:div {:style style-base/wizard-container}
+   [:div [:h3 (tr [:organization-page :heading-operator-edit])]]
+   [info/info-toggle (tr [:common-texts :instructions]) (tr [:organization-page :help-merge-company-names]) true]
+   (for [n nap-orphans
+         :let [nap-op (:transport-operator n)
+               control-disabled? false
+               ops-to-save1 (:transport-operators-to-save operator)
+               ]
```
Comment (rckopteri) [inline ote/src/cljs/ote/views/transport_operator_ytj.cljs]: trailing parentheses

File: ote/src/cljs/ote/views/transport_operator_ytj.cljs [hunk 5]
```diff
@@ -295,7 +343,9 @@
 (defn operator-ytj [e! {operator :transport-operator :as state}]
   (let [show-id-entry? (empty? (get-in state [:params :id]))
         show-details? (and (:transport-operator-loaded? state) (some? (:ytj-response state)))
-        form-options (operator-form-options e! state show-details?)
+        show-merge-companies? (and (pos-int? (count (get-in state [:transport-operator :ytj-orphan-nap-operators])))
+                                   (pos-int? (count (:ytj-company-names state))))
+        ;form-options (operator-form-options e! state show-details?)
```
Comment (rckopteri) [inline ote/src/cljs/ote/views/transport_operator_ytj.cljs]: Commented line can be removed

**Pull request merged**